### ISSUE TITLE
Problem with cue parsing when there is no per-track performer and a dash in track title

### ISSUE THREAD

**Issue by reuben-kagan** [flacon/flacon#108]
It seems that if there is a valid Album Performer field in the cuesheet, no per track performers, and a dash in the track title, flacon tries to over-guess and put the first part of the track title to the Artist field, thus breaking the parsing of legit cue sheets in an attempt to fix bogus ones.

This behavior is probably nice-to-have as an option, but not as a default,  as there are probably many cue sheets around that stores track performer in track title. and the other hand there are many tag editors around, that could be used for even more complicated things if need be.

[test.cue.txt](https://github.com/flacon/flacon/files/3972067/test.cue.txt)
[test1.cue.txt](https://github.com/flacon/flacon/files/3972068/test1.cue.txt)

**Comment by SokoloffA** [comment]
I started fixing the code, I even wrote a version that parse your CUE correctly.  But I remembered that I saw CUEs like:
```
PERFORMER "Various"

    TRACK 01 AUDIO
    TITLE "McAuley, Impellitteri, Kulick, Roth, Phillips, Schellen / I Want It All"

    TRACK 02 AUDIO
    TITLE "LaBrie, Friedman, Franklin, Appice / Sheer Heart Attack"
```

So we have a valid album performer tag (meaningless but valid) and real artists encoded in the track tags.  Program extract artist from the track only if **all tracks** contains separator in the performer tag. 
I thought false positives would be impossible because the separator can't meet in all tracks by chance. It looks like I'm wrong, in any case we will sometimes have false positives.

We need to estimate the chances of each kind of CUEs. Did you meet a real file with separators in all tracks? Please post it.

**Comment by reuben-kagan** [comment]
Hi,
I know at least two additional schemes for track naming when all the tracks would have a separator in the title tag:
1. in classical CDs they usually goes like 
01 Concerto RV135 - Allegro
02 Concerto RV135 - Largo
03 Concerto RV136 - Allegro
and so on.

2.in folk music when the sets of shorter tunes are played together in one tracks like
01 Reel 1 - Reel 2 - Reel 3
02 Jig 1 -Jig 2 - Jig 3
03 Reel 4 - Reel 5 - Reel 6 

Both are rather frequent.

**Comment by msdobrescu** [comment]
I confirm it. I'd say if some heuristic technique is applied, that should be optional by a setting.

**Comment by SokoloffA** [comment]
Probably I'll add an option for enabling/disabling heuristic analysis. By default, it will be enabled for backward compatibility.

**Comment by msdobrescu** [comment]
Please add a page in the settings if there are many decisions that may be treated separately. So far, nothing else hit me, so it works for me in other aspects.

**Comment by msdobrescu** [comment]
I'd add a list of characters that are used to split in a textbox, maybe use them in that order. Some other split characters may be also ";" or ":". If empty, no heuristics. I'd add that to the general page if it's too much to add a new heuristics page.

**Comment by SokoloffA** [comment]
@msdobrescu 

Do you think this text is understandable to an normal user?

The text of the checkbox: 

- [ ] Use heuristic definition of track performer

The hint of the checkbox: 
*"If the "TITLE" tag contains the artist's and the track names combined by a separator character, the program will try to assign the appropriate tags."*

**Comment by reuben-kagan** [comment]
sounds very clear

**Comment by msdobrescu** [comment]
@SokoloffA , I admit, I am a power user, I can't tell what a 'normal user' does. To me, although complex, EasyTag is a comprehensible app, while Kid3-qt is a mystery and I can't predict its results, but it may be only me.
Anyway, adding that text on a setting page and add to it something like "The separator characters are listed below (or here...):", followed by a text box already filled up with the ones used now, should be clear enough. But I do not insist on it. I would use it like that and add my own to it, but I don't know about others.

**Comment by SokoloffA** [comment]
I think that a field with delimiters is too much. In my opinion, the checkbox will be enough.

![Снимок экрана 2023-05-17 в 09 22 58](https://github.com/flacon/flacon/assets/854935/0e6ab72b-923d-4330-b8e8-d7bc48f5ea89)

**Comment by msdobrescu** [comment]
OK. Thanks for your patience!

**Comment by SokoloffA** [comment]
Included in release 11.0.0
